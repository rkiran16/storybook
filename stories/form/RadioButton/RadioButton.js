import { html } from "lit-html";

export const RadioButton = ({checked, isDisabled, isGroup, label}) => {
    if (!isGroup) {
        return html`
            <div class="custom-control custom-radio">
                <input type="radio" class="form-check-input" ?checked=${checked} ?disabled=${isDisabled} id="customCheck111-33" />
                <label class="form-check-label" for="customCheck111-33">${label}</label>
            </div> 
        `;
    } else {
        return html`
            <div style="width: 300px;">
                <fieldset>
                    <legend>Group Label</legend>
                    <div class="custom-control custom-radio">
                        <input type="radio" class="form-check-input" ?checked=${checked} ?disabled=${isDisabled} id="customCheck111-34" name="group" />
                        <label class="form-check-label" for="customCheck111-34">${label}</label>
                    </div>
                    <div class="custom-control custom-radio">
                        <input type="radio" class="form-check-input" ?checked=${checked} ?disabled=${isDisabled} id="customCheck111-35"name="group" />
                        <label class="form-check-label" for="customCheck111-35">Check this custom radio</label>
                    </div>
                    <div class="custom-control custom-radio">
                        <input type="radio" class="form-check-input" ?checked=${checked} ?disabled=${isDisabled} id="customCheck111-36" name="group"/>
                        <label class="form-check-label" for="customCheck111-36">This is a custom radio label that wraps to the next line</label>
                    </div>
                </fieldset>
            </div>
        `;
    }
    
}