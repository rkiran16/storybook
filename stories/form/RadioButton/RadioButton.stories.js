import { RadioButton } from './RadioButton';

export default {
   title: "Forms/RadioButton",
   parameters: {
      zeplinLink: "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/62a517bcf3187f14ebf6e27c",
   },
   argTypes : {
      isDisabled: {
         name: "isDisabled",
         type: { name: 'boolean', required: false},
         description: "Use when the user cannot proceed or until an input is collected",
         defaultValue: false,
         table: {
            type: { summary: 'boolean' },
            defaultValue: { summary: false },
         },
         control: { type: 'boolean'}
      },
      checked: {
         name: "checked",
         type: { name: 'boolean', required: false},
         description: "Use when the user cannot proceed or until an input is collected",
         defaultValue: false,
         table: {
            type: { summary: 'boolean' },
            defaultValue: { summary: false },
         },
         control: { type: 'boolean'}
      },
      label: {
         name: 'label',
         type: { name: 'string', required: true },
         description: "Radio button label",
         defaultValue: "Check this custom radio",
         control: { type: 'text'}
      }
   }
}

const RadioButtonTemplate = (args) => RadioButton(args);

export const Standard = RadioButtonTemplate.bind({});

export const Disabled = RadioButtonTemplate.bind({});
Disabled.args = {
   isDisabled: true
}

export const DisabledSelected = RadioButtonTemplate.bind({});
DisabledSelected.args = {
   isDisabled: true,
   checked: true
}

export const Group = RadioButtonTemplate.bind({});
Group.args = {
   isDisabled: false,
   checked: false,
   isGroup: true
}