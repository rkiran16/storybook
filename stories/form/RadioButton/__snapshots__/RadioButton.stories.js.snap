// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Forms/RadioButton Disabled 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
            <div class=\\"custom-control custom-radio\\">
                <input type=\\"radio\\" class=\\"form-check-input\\" ?checked=",
    " ?disabled=",
    " id=\\"customCheck111-33\\" />
                <label class=\\"form-check-label\\" for=\\"customCheck111-33\\">",
    "</label>
            </div> 
        ",
  ],
  "values": Array [
    false,
    true,
    "Check this custom radio",
  ],
}
`;

exports[`Storyshots Forms/RadioButton Disabled Selected 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
            <div class=\\"custom-control custom-radio\\">
                <input type=\\"radio\\" class=\\"form-check-input\\" ?checked=",
    " ?disabled=",
    " id=\\"customCheck111-33\\" />
                <label class=\\"form-check-label\\" for=\\"customCheck111-33\\">",
    "</label>
            </div> 
        ",
  ],
  "values": Array [
    true,
    true,
    "Check this custom radio",
  ],
}
`;

exports[`Storyshots Forms/RadioButton Group 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
            <div style=\\"width: 300px;\\">
                <fieldset>
                    <legend>Group Label</legend>
                    <div class=\\"custom-control custom-radio\\">
                        <input type=\\"radio\\" class=\\"form-check-input\\" ?checked=",
    " ?disabled=",
    " id=\\"customCheck111-34\\" name=\\"group\\" />
                        <label class=\\"form-check-label\\" for=\\"customCheck111-34\\">",
    "</label>
                    </div>
                    <div class=\\"custom-control custom-radio\\">
                        <input type=\\"radio\\" class=\\"form-check-input\\" ?checked=",
    " ?disabled=",
    " id=\\"customCheck111-35\\"name=\\"group\\" />
                        <label class=\\"form-check-label\\" for=\\"customCheck111-35\\">Check this custom radio</label>
                    </div>
                    <div class=\\"custom-control custom-radio\\">
                        <input type=\\"radio\\" class=\\"form-check-input\\" ?checked=",
    " ?disabled=",
    " id=\\"customCheck111-36\\" name=\\"group\\"/>
                        <label class=\\"form-check-label\\" for=\\"customCheck111-36\\">This is a custom radio label that wraps to the next line</label>
                    </div>
                </fieldset>
            </div>
        ",
  ],
  "values": Array [
    false,
    false,
    "Check this custom radio",
    false,
    false,
    false,
    false,
  ],
}
`;

exports[`Storyshots Forms/RadioButton Standard 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
            <div class=\\"custom-control custom-radio\\">
                <input type=\\"radio\\" class=\\"form-check-input\\" ?checked=",
    " ?disabled=",
    " id=\\"customCheck111-33\\" />
                <label class=\\"form-check-label\\" for=\\"customCheck111-33\\">",
    "</label>
            </div> 
        ",
  ],
  "values": Array [
    false,
    false,
    "Check this custom radio",
  ],
}
`;
